src/sim/signal.hh

```diff
@@ -94,12 +95,21 @@ class SignalSourcePort : public Port
 {
   private:
     SignalSinkPort<State> *sink = nullptr;
-    State _state = {};
+    State _state;
 
   public:
-    SignalSourcePort(const std::string &_name, PortID _id=InvalidPortID) :
-        Port(_name, _id)
-    {}
+    SignalSourcePort(const std::string &_name, PortID _id = InvalidPortID)
+        : Port(_name, _id)
+    {
+        _state = {};
+    }
+
+    SignalSourcePort(const std::string &_name, PortID _id,
```

Comment:
Related to Giacomo's comment, it would be good to add a comment on this function explaining why a non-standard `Port` constructor is needed here.

Reply:
Hi,  I have added the description in the next commit.